packages/common/src/common.ts

```diff
@@ -58,6 +59,7 @@ export class Common {
   public events: EventEmitter
 
   constructor(opts: CommonOpts) {
+    this._opts = opts
     this.events = new EventEmitter()
 
     this._chainParams = opts.chain
```

Comment:
Ah, but you are right, we need additional treatment for `chain`. But we should do here, then we can avoid doing any `copy()` changes.

Also (more important): if we do *not* here, then the initial `Mainnet` (or whatever chain) structure gets modified *in the first place*. That's bad and we should very much avoid (same for the `params` structs, I stumbled upon this kind of failure).